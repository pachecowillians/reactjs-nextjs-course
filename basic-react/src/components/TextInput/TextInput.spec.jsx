import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TextInput } from '.';

describe('<TextInput />', () => {
    const fn = jest.fn();
    it('should have a value', () => {
        render(<TextInput onChange={fn} value='' />);

        const input = screen.getByPlaceholderText(/type your search/i);

        expect(input.value).toBe('');
    });

    it('should call onChange function on each key pressed', () => {
        render(<TextInput onChange={fn} />);

        const input = screen.getByPlaceholderText(/type your search/i);

        userEvent.type(input, "some value");
        expect(input.value).toBe('some value');
        expect(fn).toHaveBeenCalledTimes("some value".length);
    });

    it('should match snapshot', () => {
        const { container } = render(<TextInput onChange={fn} />);
        expect(container).toMatchSnapshot();
    });
});
