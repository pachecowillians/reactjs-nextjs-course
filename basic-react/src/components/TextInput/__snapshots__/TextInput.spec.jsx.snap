// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<div>
  <input
    class="input"
    id="find-users"
    name="find-users"
    placeholder="Type your search"
    type="search"
    value=""
  />
</div>
`;
